op-e2e/system_test.go

```diff
@@ -1402,6 +1403,60 @@ func TestBatcherMultiTx(t *testing.T) {
 	t.Fatal("Expected at least 10 transactions from the batcher")
 }
 
+func TestBatcherConcurrentAltDARequests(t *testing.T) {
+	InitParallel(t)
+
+	cfg := DefaultSystemConfig(t)
+	cfg.DeployConfig.UseAltDA = true
+	cfg.BatcherMaxPendingTransactions = 0 // no limit on parallel txs
+	// ensures that batcher txs are as small as possible
+	cfg.BatcherMaxL1TxSizeBytes = derive.FrameV0OverHeadSize + 1 /*version bytes*/ + 1
+	cfg.BatcherBatchType = 0
+	cfg.DataAvailabilityType = flags.CalldataType
+	cfg.BatcherMaxConcurrentDARequest = 0 // no limit
+
+	// disable batcher because we start it manually below
+	cfg.DisableBatcher = true
+	sys, err := cfg.Start(t)
+	require.NoError(t, err, "Error starting up system")
+	defer sys.Close()
+
+	// make every request take 5 seconds, such that only concurrent requests will be able to make progress fast enough
+	sys.FakeAltDAServer.SetPutRequestLatency(5 * time.Second)
+
+	l1Client := sys.NodeClient("l1")
+	l2Seq := sys.NodeClient("sequencer")
+
+	// we wait for some L2 blocks to have been produced, just to make sure the sequencer is working properly
+	_, err = geth.WaitForBlock(big.NewInt(10), l2Seq, time.Duration(cfg.DeployConfig.L2BlockTime*15)*time.Second)
+	require.NoError(t, err, "Waiting for L2 blocks")
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	startingL1BlockNum, err := l1Client.BlockNumber(ctx)
+	require.NoError(t, err)
+
+	// start batch submission
+	driver := sys.BatchSubmitter.TestDriver()
+	err = driver.StartBatchSubmitting()
+	require.NoError(t, err)
+
+	totalTxCount := 0
+	// wait for up to 10 L1 blocks, expecting 10 L2 batcher txs in them.
+	// usually only 3 is required, but it's possible additional L1 blocks will be created
+	// before the batcher starts, so we wait additional blocks.
+	for i := int64(0); i < 10; i++ {
+		block, err := geth.WaitForBlock(big.NewInt(int64(startingL1BlockNum)+i), l1Client, time.Duration(cfg.DeployConfig.L1BlockTime*5)*time.Second)
+		require.NoError(t, err, "Waiting for l1 blocks")
+		totalTxCount += len(block.Transactions())
```

Comment:
Note that there are possibly other services in the background sending transactions, like the proposer or challenger. I think it would be better to only count the transactions having the batcher as sender.

Reply:
[1185bf8](https://github.com/ethereum-optimism/optimism/pull/11698/commits/1185bf87cfebab54f680984010275303cfbff8a2)
Shall I also change the other test?

Reply:
[d4362f3](https://github.com/ethereum-optimism/optimism/pull/11698/commits/d4362f34111c878a7738419164c8cf54afbcc792)